impl/src/main/java/org/jboss/weld/bootstrap/WeldStartup.java

```diff
@@ -545,7 +545,7 @@ public void endInitialization() {
             // fire @Initialized(ApplicationScoped.class) for non-web modules
             // web modules are handled by HttpContextLifecycle
             for (BeanDeploymentModule module : modules) {
-                if (!module.isWebModule()) {
+                if (!module.isWebModule() && !module.isFallbackModule()) {
```

Comment:
So we wouldn't fire an event for any non-web module if `EEModuleDescriptor` is not provided? That does not look correct.